src/Microsoft.OData.Edm/Csdl/Parsing/SchemaJsonParser.cs

```diff
@@ -0,0 +1,1564 @@
+            Debug.Assert(propertyObject.ValueKind == JsonValueKind.Object);
+
+            // The property object MAY contain the member $Kind with a string value of Property.
+            // This member SHOULD be omitted to reduce document size.
+            ValidateKind(propertyObject, context, "Property", required: false);
+
+            // It MAY contain the member $Type, $Collection, $Nullable, $MaxLength, $Unicode, $Precision, $Scale, $SRID, and $DefaultValue.
+            CsdlTypeReference typeReference = CsdlJsonParseHelper.ParseCsdlTypeReference(propertyObject, context);
+
+            string defaultValue = null;
+            IList<CsdlAnnotation> propertyAnnotations = new List<CsdlAnnotation>();
+            propertyObject.ParseAsObject(context, (propertyName, propertyValue) =>
+            {
+                switch (propertyName)
+                {
+                    case "$Kind":
+                    case "$Type":
+                    case "$Collection":
+                    case "$Nullable":
+                    case "$MaxLength":
+                    case "$Precision":
+                    case "$Scale":
+                    case "$Unicode":
+                    case "$SRID":
+                        // skip them because they are processed
+                        break;
+
+                    case "$DefaultValue":
+                        // The value of $DefaultValue is the type-specific JSON representation of the default value of the property.
+                        // For properties of type Edm.Decimal and Edm.Int64 the representation depends on the media type parameter IEEE754Compatible.
+                        // So far, ODL only suppports the string default value.
+                        defaultValue = propertyValue.ParseAsJson();
+                        break;
+
+                    default:
+                        // it MAY contain annotations.
+                        ParseCsdlAnnotation(propertyName, propertyValue, context, propertyAnnotations);
+                        break;
+                }
+            });
+
+            CsdlProperty csdlProperty = new CsdlProperty(name, typeReference, defaultValue, context.Location());
+            propertyAnnotations.ForEach(a => csdlProperty.AddAnnotation(a));
+            return csdlProperty;
+        }
+
+        /// <summary>
+        /// Parse the <see cref="JsonElement"/> to <see cref="CsdlNavigationProperty"/>.
+        /// </summary>
+        /// <param name="name">The name.</param>
+        /// <param name="navObject">The json value to parse from.</param>
+        /// <param name="jsonPath">The json path of this value.</param>
+        /// <returns>the parsed <see cref="CsdlNavigationProperty"/>.</returns>
+        internal static CsdlNavigationProperty ParseCsdlNavigationProperty(string name, JsonElement navObject, JsonParserContext context)
+        {
+            Debug.Assert(navObject.ValueKind == JsonValueKind.Object);

```

Comment:
>navObject.ValueKind == JsonValueKind.Object [](start = 25, length = 43)

as above, is this assert because we should have already validated that it is an object?